### PR TITLE
Adds alcohol bottle firebombs, improves explosion fire & gauze

### PR DESCRIPTION
**Pull request by ghost**
jizz
Featuring: codersprites
- Allows the simple construction of alcohol bottle firebombs by stuffing gauze into the neck of any glass alcohol bottle (including chaplain's flask)
- Bottled firebombs have less range than IEDs, but more chance to have fire on the turfs around the detonation.
- Added a flame_prob arg to explosions, which defaults to 40 like it did before, to un-hardcode the probability of explosions creating fire. You can now cause explosions with **hotspots on every turf in the flame_range** instead of it being hardcoded at 40 prob.
- Added a unique sprite for improvised gauze, renamed it to 'strips of cloth' since it has multiple uses now outside of just acting as gauze.
- All is_sharp weapons can cut cloth from bedsheets.
- Added a 'flammable' var to reagents. This allows the firebombs to be filled with any flammable liquid in order to cause a fireball around where they're thrown. This includes all children of ethanol, welding fuel, thermite, unholywater, etc.
- Bottled firebombs can get too hot to handle, causing you to drop them and potentially light yourself on fire. This will happen regardless of gloves, for balance's sake so you can't walk around with a primed firebomb endlessly.
- You can't drink from bottles that have gauze stuck in them. Potential counter-play for chaplain flasks as cultists? Lighting his holy water on fire and tossing it at sec?
- You can remove gauze from bottles as long as it hasn't been lit yet.
- You can light bottled firebombs with anything you can light cigarettes with.

### CODE REVIEW

File: code/modules/reagents/Chemistry-Holder.dm [hunk 1]
```diff
@@ -5,6 +5,9 @@ var/const/INGEST = 2
 
 ///////////////////////////////////////////////////////////////////////////////////
 
+datum/reagent //why are these named so similarly ugh
+	var/flammable = 0
```
Comment (Iamgoofball) [inline code/modules/reagents/Chemistry-Holder.dm]: just add this to Chemistry-Reagents.dm

Comment (paprka) [inline code/modules/reagents/Chemistry-Holder.dm]: Aight

**Comment by WJohn** [comment]
It'd be nice if it extended to medical drapes, cloths used for the bar, and possibly even animal hides for this. Otherwise :+1:

**Comment by Iamgoofball** [comment]
:+1: x :100:

i'll update the goonchems with these vars as appropriate when this gets merged

**Comment by paprka** [comment]
flammability with chems can probably be extended to other things, but for right now flammable reagents just add fire_stacks to mobs hat they get splashed on, which is probably easier in the end. I would like turfs to be flammable too, so you can spill welding fluid on a bunch of turfs and then light them on fire to create a fire wall or burn down medbay with a tank of fuel

**Comment by Jordie0608** [comment]
http://keepachangelog.com/

**Comment by RandomMarine** [comment]
>This will happen regardless of gloves, for balance's sake so you can't walk around with a primed firebomb endlessly.

Black gloves need actual uses to make people consider taking them instead of insulated. This would be a start if you allowed them to protect.

**Comment by Razharas** [comment]
The way you handle overheating is strange
Isnt it easier to just spawn(rand(50,70)), check if its still lit and then boom?
Adding it to process and checking its lit each tick with each time with random time delta and never removing it

**Comment by paprka** [comment]
Meh, I'm not a master of convenient nor efficient code really

I'll do your suggestion and possibly add a check for gloves I guess

**Comment by Razharas** [comment]
Thats better

**Comment by Atskadan** [comment]
i was actually going to finish my version of this once i got unbanno'd but nice
why arent they called molotovs tho

**Comment by Iamgoofball** [comment]
>merge conflicts

we're fucked, rip paprika

**Comment by Cheridan** [comment]
He can still push commits to open PRs. Or a maintainer can do a manual merge.

That said... we already have IEDs. So these are just redundant.
"But IEDs suck, Cheridan"
Then buff 'em, dingus.

**Comment by Atskadan** [comment]
ieds are for breaking shit and hurting people
these are for setting people on fire

**Comment by optimumtact** [comment]
IED's are basically exactly these now, since their explosive form was just a pain for everyone.

**Comment by Cheridan** [comment]
Gonna go ahead close this based on redundancy. There's also some other minor issues I don't like (attackbys instead of muh tablecrafting, etc) and the author isn't really around to handle them.

**Pull request closed without merging**